### PR TITLE
feat: Elevator evergreen content

### PR DESCRIPTION
**Pull request by cmaddox5**
Adds evergreen content support on elevator screens. Assuming `Screens.Util.Assets.s3_asset_url/1` points at the `screens-dev-green` bucket, you can use the following config to test:

```json
{
  "ELE-101": {
    "disabled": false,
    "name": "",
    "app_id": "elevator_v2",
    "app_params": {
      "evergreen_content": [
        {
          "priority": [0],
          "slot_names": ["header"],
          "asset_path": "images/elevator/header_test.png",
          "schedule": [
            {
              "start_dt": null,
              "end_dt": null
            }
          ],
          "text_for_audio": null,
          "audio_priority": null
        },
        {
          "priority": [0],
          "slot_names": ["main_content"],
          "asset_path": "images/elevator/main_content_test.png",
          "schedule": [
            {
              "start_dt": null,
              "end_dt": null
            }
          ],
          "text_for_audio": null,
          "audio_priority": null
        },
        {
          "priority": [0],
          "slot_names": ["footer"],
          "asset_path": "images/elevator/footer_test.png",
          "schedule": [
            {
              "start_dt": null,
              "end_dt": null
            }
          ],
          "text_for_audio": null,
          "audio_priority": null
        }
      ],
      "elevator_id": "842",
      "alternate_direction_text": "Alternate direction text goes here. This is the largest body text class. Use this for shorter sets of directions. Add up to 6 lines at this size",
      "accessible_path_image_url": "images/elevator/maps/forest-hills-map.png",
      "accessible_path_direction_arrow": "e",
      "accessible_path_image_here_coordinates": {
        "x": 151,
        "y": 182
      }
    },
    "device_id": "",
    "hidden_from_screenplay": false,
    "refresh_if_loaded_before": null,
    "tags": [],
    "vendor": "mimo"
  }
}
```

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1208310396563613

### CODE REVIEW

File: lib/screens/v2/candidate_generator/elevator.ex [hunk 1]
```diff
@@ -27,10 +28,11 @@ defmodule Screens.V2.CandidateGenerator.Elevator do
   def candidate_instances(
         config,
         now \\ DateTime.utc_now(),
-        elevator_closure_instances_fn \\ &ElevatorClosures.elevator_status_instances/1
+        elevator_closure_instances_fn \\ &ElevatorClosures.elevator_status_instances/1,
+        evergreen_content_instances_fn \\ &Evergreen.evergreen_content_instances/2
       ) do
     [header_instance(config, now), footer_instance(config)] ++
-      elevator_closure_instances_fn.(config)
+      elevator_closure_instances_fn.(config) ++ evergreen_content_instances_fn.(config, now)
```
[the remark this answers is not in the document]
Comment (cmaddox5) [reply lib/screens/v2/candidate_generator/elevator.ex:35]: That's a good idea. Will change that now.

File: lib/screens/v2/candidate_generator/elevator.ex [hunk 2]
```diff
@@ -27,10 +28,15 @@ defmodule Screens.V2.CandidateGenerator.Elevator do
...
         now \\ DateTime.utc_now(),
-        elevator_closure_instances_fn \\ &ElevatorClosures.elevator_status_instances/1
+        elevator_closure_instances_fn \\ &ElevatorClosures.elevator_status_instances/1,
+        evergreen_content_instances_fn \\ &Evergreen.evergreen_content_instances/2
       ) do
-    [header_instance(config, now), footer_instance(config)] ++
-      elevator_closure_instances_fn.(config)
+    Enum.concat([
+      header_instance(config, now),
+      footer_instance(config),
```
Comment (digitalcora) [inline lib/screens/v2/candidate_generator/elevator.ex:36]: I'm confused as to how this works. `header_` and `footer_instance` return a single widget struct, which is not an enumerable, and `Enum.concat/1` expects an enum _of_ enums. Doing the equivalent of this call in IEx results in an error:

```
iex(1)> Enum.concat([:a, :b, [:c, :d], [:e, :f]])
** (Protocol.UndefinedError) protocol Enumerable not implemented for :a of type Atom
    (elixir 1.15.7) lib/enum.ex:1: Enumerable.impl_for!/1
    (elixir 1.15.7) lib/enum.ex:166: Enumerable.reduce/3
    (elixir 1.15.7) lib/enum.ex:4387: anonymous fn/3 in Enum.concat_enum/1
    (elixir 1.15.7) lib/enum.ex:2510: Enum."-concat_enum/1-lists^foldl/2-1-"/3
    (elixir 1.15.7) lib/enum.ex:2510: Enum.concat_enum/1
    iex:1: (file)
```

Comment (digitalcora) [reply lib/screens/v2/candidate_generator/elevator.ex:36]: [\*cough\*](https://mbtace.sentry.io/issues/6092494293)

Comment (cmaddox5) [reply lib/screens/v2/candidate_generator/elevator.ex:36]: 🫥

**Pull request merged**